providers/structured/provider.go

```diff
@@ -34,12 +35,21 @@ type StructuredOutputLogProvider struct {
 	rawLogCalls      map[providers.RawLogCallType][]RawLogCallArgs
 	recordCalls      []RecordCallArgs
 	recordEventCalls []RecordEventCallArgs
+	logMutex         sync.RWMutex
+	metricMutex      sync.Mutex
+	eventMutex       sync.Mutex
 }
 
 func (p StructuredOutputLogProvider) GetRawLogCalls() map[providers.RawLogCallType][]RawLogCallArgs {
```

Comment:
I’d vote for removing this method now; it shouldn’t be too much trouble to switch to GetRawLogCallsByCallType.

Reply:
Ok I'll do that and add to the change log that this is a breaking change.